src/app/plugins/tictactoe.plugin.ts

```diff
@@ -192,19 +192,24 @@ class TTTGame {
 
         // Make the move and evaluate the board state
         this._board[row][col] = this.currentPlayer;
-        moves.push({ row, col, val: this._evaluate(this.currentPlayer * -1) });
+        moves.push({ row, col, val: this._minimax(this.currentPlayer * -1) });
         // Backtrack.
         this._board[row][col] = 0;
       })
     );
 
     // Sort the moves in decreasing value
     moves.sort((a, b) => b.val - a.val);
-    return { bestRow: moves[0].row, bestCol: moves[0].col };
+
+    // Calculate all moves with equal value, and return one.
+    const bestMoves = moves.filter(move => move.val === moves[0].val);
```

Comment:
Is it guaranteed that `moves.length !== 0`?

Reply:
Yes

Reply:
This function only runs if the game is not over.